pkg/nmi/server/server.go

```diff
@@ -362,14 +366,16 @@ func (s *Server) msiHandler(w http.ResponseWriter, r *http.Request) (ns string)
 	podID, err = s.TokenClient.GetIdentities(r.Context(), podns, podname, rqClientID)
 	if err != nil {
 		klog.Errorf("failed to get matching identities for pod: %s/%s, error: %+v", podns, podname, err)
-		http.Error(w, err.Error(), getErrorResponseStatusCode(podID != nil))
+		stausCode = getErrorResponseStatusCode(podID != nil)
+		http.Error(w, err.Error(), stausCode)
 		return
 	}
 
 	var token *adal.Token
 	token, err = s.TokenClient.GetToken(r.Context(), rqClientID, rqResource, *podID)
 	if err != nil {
 		klog.Errorf("failed to get service principal token for pod: %s/%s, error: %+v", podns, podname, err)
+		stausCode = http.StatusInternalServerError
```

Comment:
Here I use StatusInternalServerError since I would like to consider in this case, it should be a Cosmic infra issue instead of partner's issue. StatusForbidden is a used to indicate partner's issue. Let me know if  you have any concerns on this.

Reply:
Seems fine to me

---
In reply to: [448077153](https://github.com/bowu24/aad-pod-identity/pull/8#discussion_r448077153) [](ancestors = 448077153)

Reply:
cool. Thanks